### PR TITLE
Week 4 Charts

### PR DESCRIPTION
**Pull request by nevingeorge**
This finishes up the charts tutorial, which is the final tutorial of week 4. I added a timeline of places I've lived to the gallery page, and if you log in on the contact page you can vote on which project I should work on next. The project vote chart stores the results in datastore. To add the chart to the contact page, I also had to restructure and modularize contact.html (should be cleaner).

### CODE REVIEW

**Comment by nevingeorge** [comment]
In order to fix a bug where users can post comments and vote without having set a name, I had to delete everyone's account information. Therefore, if you go to the website again you will most likely be asked to re-enter your name (which is good, because then you can see what I implemented to fix the bug).

File: portfolio/src/main/java/com/google/sps/servlets/NameServlet.java [hunk 1]
```diff
@@ -104,4 +111,14 @@ public static String getLastName(String id) {
     String lastName = (String) entity.getProperty("lastName");
     return lastName;
   }
+
+  // returns whether a user has voted
+  public static String getHasVoted(String id) {
```
Comment (avenkat-google) [inline portfolio/src/main/java/com/google/sps/servlets/NameServlet.java:111]: Why not make this a boolean so the output is more clear? A string can have many different values.

File: portfolio/src/main/java/com/google/sps/servlets/NextProjectServlet.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    UserService userService = UserServiceFactory.getUserService();
+    String id = userService.getCurrentUser().getUserId();
+    String firstName = NameServlet.getFirstName(id);
+    String lastName = NameServlet.getLastName(id);
+
+    Entity entity = new Entity("UserInfo", id);
+    entity.setProperty("id", id);
+    entity.setProperty("firstName", firstName);
+    entity.setProperty("lastName", lastName);
+    entity.setProperty("voted", "cannotVote");
```
Comment (avenkat-google) [inline portfolio/src/main/java/com/google/sps/servlets/NextProjectServlet.java:72]: Just a suggestion, but would be nice to be able to change my vote.

**Comment by nevingeorge** [comment]
I'm sorry about this last commit - I went to make the output of getHasVoted boolean like you suggested, and to do that I realized I would have to make some changes regarding how I'm storing the user's info. I thought then that the best way to implement these necessary changes was to create the user info class that you suggested. In order to create this class, I had to restructure essentially all of the code. The code is much clearer now; however, I didn't expect to have to change everything in one commit.

**Pull request merged**